### PR TITLE
DOC/MNT: Throwing some docstrings at axes_rgb.py

### PR DESCRIPTION
**Pull request by ericdill**
In my learning how to use axes_rgb.py, I ended up writing some
docstrings. Hope this is useful.

### CODE REVIEW

File: lib/mpl_toolkits/axes_grid1/mpl_axes.py [hunk 1]
```diff
@@ -33,10 +33,12 @@ def __init__(self, axes):
 
         def __getitem__(self, k):
             if isinstance(k, tuple):
-                r = SimpleChainedObjects([dict.__getitem__(self, k1) for k1 in k])
+                r = SimpleChainedObjects(
+                    [super(Axes.AxisDict, self).__getitem__(self, k1)
```
Comment (tacaswell) [inline lib/mpl_toolkits/axes_grid1/mpl_axes.py]: `super(cls, self)` returns the bound method so you don't need the second self.

Comment (ericdill) [inline lib/mpl_toolkits/axes_grid1/mpl_axes.py]: Couldn't I also just do `super(Axes.AxisDict, self)[k1]` here? dict subclassing is still a mystery to me

Comment (tacaswell) [inline lib/mpl_toolkits/axes_grid1/mpl_axes.py]: Yes, `super()[k]` would also work.

Comment (ericdill) [inline lib/mpl_toolkits/axes_grid1/mpl_axes.py]: Any preference?

File: lib/mpl_toolkits/axes_grid1/axes_rgb.py [hunk 2]
```diff
@@ -75,17 +77,59 @@ def imshow_rgb(ax, r, g, b, **kwargs):
...
-        if axes_class is None:
-            axes_class = self._defaultAxesClass
+        try:
+            axes_class = kwargs.pop("axes_class", self._defaultAxesClass)
+        except AttributeError:
+            new_msg = ("A subclass of RGBAxesBase must have a "
+                       "_defaultAxesClass attribute. If you are not sure which "
+                       "axes class to use, consider using "
+                       "mpl_toolkits.axes_grid1.mpl_axes.Axes.")
+            six.reraise(KeyError, AttributeError(new_msg),
```
Comment (tacaswell) [inline lib/mpl_toolkits/axes_grid1/axes_rgb.py]: Please do not change the exception type, this is an API break.

Comment (ericdill) [inline lib/mpl_toolkits/axes_grid1/axes_rgb.py]: Good catch... That was not intentional

File: lib/mpl_toolkits/axes_grid1/axes_rgb.py [hunk 3]
```diff
@@ -72,20 +73,60 @@ def imshow_rgb(ax, r, g, b, **kwargs):
...
-
-
-
-
-        if axes_class is None:
-            axes_class = self._defaultAxesClass
+        try:
+            axes_class = kwargs.pop("axes_class", self._defaultAxesClass)
+        except AttributeError:
+            new_msg = ("A subclass of RGBAxesBase must have a "
```
Comment (pelson) [inline lib/mpl_toolkits/axes_grid1/axes_rgb.py]: hmmm. ```_defaultAxesClass``` *is* private, in general I'm not in favour of this defensive style, but I accept that this message would have been very helpful for you at some point. I wont hold the PR up for this though.

**Pull request merged**